### PR TITLE
[AsyncMessages]Channel, Thread and ChannelList updates

### PR DESCRIPTION
**Pull request by ipavlidakis**
**Stack**:
- #2623
- #2622 ⬅
- #2621
- #2620
- #2619
- #2618

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

**Review by nuno-vieira — APPROVED** [review]
LGTM! ✅

File: Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift [hunk 1]
```diff
@@ -250,4 +282,8 @@ extension ChatChannelListItemView {
 
         return names + " \(users.count == 1 ? typingSingularText : typingPluralText)"
     }
+
+    var lastMessageIsVoiceRecording: Bool {
```
Comment (polqf) [inline Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift:286]:
```suggestion
    var isLastMessageVoiceRecording: Bool {
```

**Review by polqf — COMMENTED** [review]
These snapshots seem to be producing issues:

[test_appearanceCustomization_usingAppearance.rightToLeftLayout-default.png](https://github.com/GetStream/stream-chat-swift/pull/2622/files#diff-0b98f05a4805257136cf199e4902aa989bea625d900d2c720fb6099cac4ee0ed)
[test_appearanceCustomization_usingSubclassing.rightToLeftLayout-default.png](https://github.com/GetStream/stream-chat-swift/pull/2622/files#diff-97a97841489b8b8f7205c73473006bce07811b09771114902ff3bf65f59a5f64)
[test_defaultAppearance.rightToLeftLayout-default.png](https://github.com/GetStream/stream-chat-swift/pull/2622/files#diff-7d3da821ddca3f211967785745ef519420d1dbf824a72307933390bcc21a2a94)

**Comment by ipavlidakis** [comment]
> test_defaultAppearance.rightToLeftLayout-default.png

I think that the new snapshots just highlight an existing issue here. As it is right to left, the preview message should be on the right side (next to the avatar). The problem i see is the channel title that's wrong. WDYT @nuno-vieira ?

**Comment by ipavlidakis** [comment]
✓ Commit merged in pull request [#2623](https://github.com/GetStream/stream-chat-swift/pull/2623)

**Pull request closed without merging**